crates/sui/src/client_commands.rs

```diff
@@ -1280,6 +1302,28 @@ impl SuiClientCommands {
                     .collect();
                 SuiClientCommandResult::Gas(coins)
             }
+            SuiClientCommands::Faucet { address, url } => {
+                let address = get_identity_address(address, context)?;
+                let url = if let Some(url) = url {
+                    url
+                } else {
+                    let active_env = context.config.get_active_env();
+
+                    if let Ok(env) = active_env {
+                        let network = match env.rpc.as_str() {
+                            SUI_DEVNET_URL => "https://faucet.devnet.sui.io/v1/gas",
+                            SUI_TESTNET_URL => "https://faucet.testnet.sui.io/v1/gas",
+                            SUI_LOCAL_NETWORK_URL => SUI_LOCAL_NETWORK_GAS_URL,
+                            _ => bail!("Cannot recognize the active network. Please provide the gas faucet full URL.")
+                        };
+                        network.to_string()
+                    } else {
+                        bail!("No URL for faucet was provided and there is no active network.")
+                    }
+                };
```

Comment:
nit: It looks like this can be coalesced into a single `if ... else if`:

```suggestion
                } else if let Ok(env) = context.config.get_active_env() {
                    match env.rpc.as_str() {
                        SUI_DEVNET_URL => "https://faucet.devnet.sui.io/v1/gas",
                        SUI_TESTNET_URL => "https://faucet.testnet.sui.io/v1/gas",
                        SUI_LOCAL_NETWORK_URL => SUI_LOCAL_NETWORK_GAS_URL,
                        _ => bail!("Cannot recognize the active network. Please provide the gas faucet full URL.")
                    }
                    .to_string()
                } else {
                    bail!("No URL for faucet was provided and there is no active network.")
                };
```